app/components/documents/document-preview-sidebar.js

```diff
@@ -23,13 +39,88 @@ export default class DocumentsDocumentPreviewSidebar extends Component {
 
   @task
   *loadData() {
-    const docContainer = yield this.args.piece.documentContainer;
-    this.documentType = yield docContainer.type;
+    this.docContainer = yield this.args.piece.documentContainer;
+    this.loadVersionHistory.perform();
+    this.documentType = yield this.docContainer.type;
     this.accessLevel = yield this.args.piece.accessLevel;
   }
 
+  @task
+  *loadVersionHistory() {
+    this.versionPieces = yield this.docContainer.reverseSortedPieces;
+    this.lastPiece = yield this.docContainer.lastPiece;
+  }
+
   @action
   setActiveTab(tabName) {
     this.activeTab = tabName;
   }
+
+  @action
+  openUploadVersionModal() {
+    this.isOpenUploadVersionModal = true;
+  }
+
+  @action
+  closeUploadVersionModal() {
+    this.isOpenUploadVersionModal = false;
+  }
+
+  @action
+  async saveUploadVersionModal(newVersion) {
+    let accessLevel = await this.lastPiece.accessLevel;
+    if (isBlank(accessLevel)){
+      accessLevel = await this.store.findRecordByUri('access-level', CONSTANTS.ACCESS_LEVELS.INTERN_REGERING);
+    }
+    const now = moment().utc()
+      .toDate();
+    let newPiece = this.store.createRecord('piece', {
+      created: now,
+      modified: now,
+      name: newVersion.name,
+      file: newVersion.file,
+      previousPiece: this.lastPiece,
+      confidential: this.lastPiece.confidential,
+      accessLevel: accessLevel,
+      documentContainer: this.docContainer,
+    });
+    await newPiece.save();
+
+    this.isOpenUploadVersionModal = false;
+    this.loadVersionHistory.perform();
+    this.router.transitionTo('document', newPiece.id)
```

Comment:
Het is niet de bedoeling om steeds een nieuwe tab te openen wanneer de gebruiker een andere versie wil bekijken maar om de geselecteerde versie in de huidige preview up te daten.